src/game/Warden/Warden.h

```diff
@@ -95,6 +95,43 @@ struct WardenHashRequest
     uint8 Seed[16];
 };
 
+namespace WardenState
+{
+    enum Value
+    {
+        STATE_INITIAL,
+        STATE_REQUESTED_MODULE,
+        STATE_SENDED_MODULE,
+        STATE_REQUESTED_HASH,
+        STATE_INITIALIZE_MODULE,
+        STATE_REQUESTED_DATA,
+        STATE_RESTING
+    };
+
+    inline char* to_string(WardenState::Value value)
+    {
+        switch (value)
+        {
+            case WardenState::STATE_INITIAL:
+                return "STATE_INITIAL";
+            case WardenState::STATE_REQUESTED_MODULE:
+                return "STATE_REQUESTED_MODULE";
+            case WardenState::STATE_SENDED_MODULE:
+                return "STATE_SENDED_MODULE";
+            case WardenState::STATE_REQUESTED_HASH:
+                return "STATE_REQUESTED_HASH";
+            case WardenState::STATE_INITIALIZE_MODULE:
+                return "STATE_INITIALIZE_MODULE";
+            case WardenState::STATE_REQUESTED_DATA:
+                return "STATE_SENDED_DATA";
+            case WardenState::STATE_RESTING:
+                return "STATE_RESTING";
+            default:
+                return "UNDEFINED STATE";
```

Comment:
I think a better way is to leave the default case out and have a return in the end. This is because compilers can detect and complain if you have not defined all cases if you dont have a default case and we could maybe see if there is a missing warden tostring definition. But thats my opinion.